File: app/src/lib/job-status.ts
export const DISPATCH_JOB_STATUS = {
  UNKNOWN: 'UNKNOWN',
  ADMIN_PREVIEW: 'ADMIN_PREVIEW',
  READY_FOR_TECH: 'READY_FOR_TECH',
  PENDING_ADMIN_CONFIRMATION: 'PENDING_ADMIN_CONFIRMATION',
  PENDING_REVIEW: 'PENDING_REVIEW',
  PENDING: 'PENDING',
  SCHEDULED: 'SCHEDULED',
  IN_PROGRESS: 'IN_PROGRESS',
  DELAYED: 'DELAYED',
  COMPLETED: 'COMPLETED',
  CANCELLED: 'CANCELLED',
} as const;

export type DispatchJobStatus = typeof DISPATCH_JOB_STATUS[keyof typeof DISPATCH_JOB_STATUS];

const STATUS_ALIASES: Record<string, DispatchJobStatus> = {
  admin_review: DISPATCH_JOB_STATUS.ADMIN_PREVIEW,
  admin_preview: DISPATCH_JOB_STATUS.ADMIN_PREVIEW,
  ready_for_tech: DISPATCH_JOB_STATUS.READY_FOR_TECH,
  ready_for_tech_acceptance: DISPATCH_JOB_STATUS.PENDING,
  pending_admin_confirmation: DISPATCH_JOB_STATUS.PENDING_ADMIN_CONFIRMATION,
  pending_review: DISPATCH_JOB_STATUS.PENDING_REVIEW,
  pending: DISPATCH_JOB_STATUS.PENDING,
  scheduled: DISPATCH_JOB_STATUS.SCHEDULED,
  in_progress: DISPATCH_JOB_STATUS.IN_PROGRESS,
  delayed: DISPATCH_JOB_STATUS.DELAYED,
  completed: DISPATCH_JOB_STATUS.COMPLETED,
  cancelled: DISPATCH_JOB_STATUS.CANCELLED,
};

export function normalizeDispatchJobStatus(value: string | null | undefined): DispatchJobStatus {
  const normalized = (value || '').trim().toLowerCase();
  return STATUS_ALIASES[normalized] ?? DISPATCH_JOB_STATUS.UNKNOWN;
}

File: app/backend/app/core/job_status.py
from enum import Enum
from typing import Optional


class DispatchJobStatus(str, Enum):
    UNKNOWN = "UNKNOWN"
    ADMIN_PREVIEW = "ADMIN_PREVIEW"
    READY_FOR_TECH = "READY_FOR_TECH"
    PENDING_ADMIN_CONFIRMATION = "PENDING_ADMIN_CONFIRMATION"
    PENDING_REVIEW = "PENDING_REVIEW"
    PENDING = "PENDING"
    SCHEDULED = "SCHEDULED"
    IN_PROGRESS = "IN_PROGRESS"
    DELAYED = "DELAYED"
    COMPLETED = "COMPLETED"
    CANCELLED = "CANCELLED"


_NORMALIZATION_MAP = {
    "admin_review": DispatchJobStatus.ADMIN_PREVIEW,
    "admin_preview": DispatchJobStatus.ADMIN_PREVIEW,
    "ready_for_tech": DispatchJobStatus.READY_FOR_TECH,
    "ready_for_tech_acceptance": DispatchJobStatus.PENDING,
    "pending_admin_confirmation": DispatchJobStatus.PENDING_ADMIN_CONFIRMATION,
    "pending_review": DispatchJobStatus.PENDING_REVIEW,
    "pending": DispatchJobStatus.PENDING,
    "scheduled": DispatchJobStatus.SCHEDULED,
    "in_progress": DispatchJobStatus.IN_PROGRESS,
    "delayed": DispatchJobStatus.DELAYED,
    "completed": DispatchJobStatus.COMPLETED,
    "cancelled": DispatchJobStatus.CANCELLED,
}


def normalize_dispatch_job_status(value: Optional[str]) -> DispatchJobStatus:
    normalized = (value or "").strip().lower()
    return _NORMALIZATION_MAP.get(normalized, DispatchJobStatus.UNKNOWN)


def normalize_status(value: Optional[str]) -> DispatchJobStatus:
    return normalize_dispatch_job_status(value)


def db_status_from_dispatch_status(value: DispatchJobStatus) -> str:
    if value == DispatchJobStatus.ADMIN_PREVIEW:
        return "admin_review"
    return value.value.lower()
